proxy/gateway/java/gateway-remote-module-sdk/src/main/java/com/microsoft/azure/gateway/remote/MessageDeserializer.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ */
+package com.microsoft.azure.gateway.remote;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Deserializer for a Gateway message. The messages from the Gateway are received as an array of bytes in a specific format. 
+ * 
+ */
+class MessageDeserializer {
+
+    // 0xA1 comes from (A)zure (I)oT
+    private static final byte FIRST_MESSAGE_BYTE = (byte) 0xA1;
+    // 0x6C comes from (G)ateway control message
+    private static final byte SECOND_MESSAGE_BYTE = (byte) 0x6C;
+    private static final byte BASE_MESSAGE_SIZE = 8;
+    private static final byte BASE_CREATE_SIZE = BASE_MESSAGE_SIZE + 10;
+
+    /**
+     * Deserializes the message and constructs a {@link ControlMessage}.
+     *
+     * @param messageBuffer The message content
+     * @param version The message version that should be parsed.
+     *
+     * @return
+     * @throws MessageDeserializationException If the message is malformed.
+     */
+    public RemoteMessage deserialize(ByteBuffer messageBuffer, byte version) throws MessageDeserializationException {
+        RemoteMessageType msgType = null;
+        int totalSize = 0;
+        messageBuffer.position(0);
+
+        if (messageBuffer.limit() < BASE_MESSAGE_SIZE) {
+            throw new MessageDeserializationException(
+                    String.format("Message size %s should be >= %s", messageBuffer.limit(), BASE_MESSAGE_SIZE));
+        }
+
+        byte header1 = messageBuffer.get();
+        byte header2 = messageBuffer.get();
+        if (header1 == FIRST_MESSAGE_BYTE && header2 == SECOND_MESSAGE_BYTE) {
+            byte messageVersion = messageBuffer.get();
+            if (messageVersion > version)
+                throw new MessageDeserializationException(String.format("Message version %s can not be higher than configured version %s", messageVersion, version));
+
+            byte type = messageBuffer.get();
```

Comment:
I talked with Dave and it is sent as a byte in the message. On the gateway side it converts it to int